packages/calypso-e2e/src/lib/blocks/block-flows/subscribe.ts

```diff
@@ -19,7 +19,12 @@ export class SubscribeFlow implements BlockFlow {
 		const emailInputLocator = context.page
 			.getByRole( 'main' )
 			.getByRole( 'textbox', { name: 'Type your email' } );
-		await emailInputLocator.fill( 'foo@example.com' );
+
+		// The email input field only appears if not logged in, or if logged in user is not subscribed already.
+		if ( emailInputLocator ) {
+			await emailInputLocator.fill( 'foo@example.com' );
+		}
```

Comment:
This condition is broken, because it will always be true. The locator instance is always created as locators are lazily evaluated. The correct way of checking if an element is present would be, for example, checking if it's visible:

```js

if ( await emailInputLocator.isVisible() ) {
	await emailInputLocator.fill( 'foo@example.com' );
}

Reply:
Thanks for double checking! Just noting this was noted and fixed in https://github.com/Automattic/wp-calypso/pull/83883